index.js

```diff
@@ -7,440 +8,217 @@
-        function done(error, results) {
-            if (callback) {
-                callback(error, results);
-            } else {
-                error ? reject(error) : resolve(results);
-            }
-        }
-
-        let reqOpts = {url: urlOpts.url};
-        if (responseType === RESPONSE_TYPE.ZIP) {
-            reqOpts.encoding = null;
+    getUpdates(fromTime, toTime) {
+        let uri = `updated/query?fromTime=${fromTime}`;
+        if (toTime) {
+            uri = `${uri}&toTime=${toTime}`;
         }
+        return this.requestGet({uri});
+    }
 
-        request(reqOpts, function(error, resp, data) {
-            if (!responseOk(error, resp, data)) {
-                if (!error) {
-                    error = new Error("Could not complete the request");
-                }
-                error.statusCode = resp ? resp.statusCode : undefined;
+    /**
+     * @param {Number|String} seriesId
+     * @param {String} [language]
+     * @returns {Promise}
+     */
+    getSeriesAllById(seriesId, language) {
+        return Promise.all([
+            this.getSeriesById(seriesId, language),
+            this.getEpisodesBySeriesId(seriesId, language)
+        ])
+            .then(results => {
+                let series = results[0];
+                series.Episodes = results[1];
+                return series;
+            });
+    }
 
-                return done(error);
-            } else if (error) {
-                return done(error);
+    /**
+     * @private
+     */
+    login() {
+        this.loginPromise = request.post({
+            baseUrl,
+
+            json: true,
+            uri: 'login',
+            body: {
+                apikey: this.apiKey
             }
-
-            if (responseType === RESPONSE_TYPE.ZIP) {
-
-                Zip.loadAsync(data).then(function(zip) {
-                    return zip.file(`${urlOpts.lang}.xml`).async("string")
-                }).then(function(contents) {
-                    parseXML(contents, normalise, done);
-                }).catch(function(error) {
-                    done(error);
+        })
+            .then(data => data.token)
+            .then(token => {
+                this.defaultRequest = request.defaults({
+                    baseUrl,
+
+                    json: true,
+                    headers: {
+                        "User-Agent": "edwellbrook/node-tvdb",
+                        Authorization: `Bearer ${token}`,
+                        'Accept-language': this.language
+                    }
                 });
+            });
+    }
 
-            } else {
-                parseXML(data, normalise, done);
-            }
-        });
-    });
-}
-
-/**
- * Parse XML response
- *
- * @param {String} xml data
- * @param {Function} normalise - a function to tidy the response object
- * @param {Function} callback
- * @api private
- */
+    /**
+     * @private
+     */
+    requestGet(options, language) {
```

Comment:
Private methods still need descriptions and parameters listed in the jsdoc comment